code/game/objects/items/storage/firstaid.dm

```diff
@@ -318,15 +318,21 @@
 /obj/item/storage/pill_bottle
 	name = "pill bottle"
 	desc = "It's an airtight container for storing medication."
-	icon_state = "pill_canister"
+	icon_state = "pill_canister_0"
 	icon = 'icons/obj/chemical.dmi'
 	item_state = "contsolid"
 	lefthand_file = 'icons/mob/inhands/equipment/medical_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/medical_righthand.dmi'
 	w_class = WEIGHT_CLASS_SMALL
+	var/pill_variance = 100 //probability pill_bottle has a different icon state. Put at 0 for no variance
+	var/pill_type = "pill_canister_"
 
 /obj/item/storage/pill_bottle/ComponentInitialize()
+	var/proper_name = name
 	. = ..()
+	name = proper_name
+	if(prob(pill_variance))
+		icon_state = "[pill_type][rand(0,6)]"
```

Comment:
ew non component things in comopnentinitialize

Reply:
What bacon means is put these changes in regular Initialize()

Reply:
Ah alright, will do